packages/realm/src/ProgressRealmPromise.ts

```diff
@@ -158,6 +171,48 @@ export class ProgressRealmPromise implements Promise<Realm> {
     }
   };
 
+  private createTimeoutPromise(config: Configuration, { timeOut, timeOutBehavior }: OpenBehavior) {
+    if (typeof timeOut === "number") {
+      this.timeoutPromise = new TimeoutPromise(
+        this.handle.promise, // Ensures the timeout gets cancelled when the realm opens
+        timeOut,
+        `Realm could not be downloaded in the allocated time: ${timeOut} ms.`,
+      );
+      if (timeOutBehavior === OpenRealmTimeOutBehavior.ThrowException) {
+        // Make failing the timeout, reject the promise
+        this.timeoutPromise.catch(this.handle.reject);
+      } else if (timeOutBehavior === OpenRealmTimeOutBehavior.OpenLocalRealm) {
+        // Make failing the timeout, resolve the promise
+        this.timeoutPromise.catch((err) => {
+          if (err instanceof TimeoutError) {
+            const realm = new Realm(config);
+            this.cancelAndResetTask();
+            this.handle.resolve(realm);
```

Comment:
Question, does the realm need to be created before the call to `cancelAndResetTask()`?

Reply:
Not necessarily, I'd think the order doesn't matter. It would probably be cleaner to turn that around and group the creation of the `Realm` with resolving the `PromiseHandle` 👍 
```suggestion
            this.cancelAndResetTask();
            const realm = new Realm(config);
            this.handle.resolve(realm);
```